app/service.js

```diff
@@ -20,31 +20,38 @@ function buildObj(pairs) {
   }, {});
 }
 
+function makeUrlAbsolute(base, relative) {
+  const relativeParsed = urlparse.parse(relative);
+
+  if (relativeParsed.host === null) {
+    return urlparse.resolve(base, relative);
+  }
+
+  return relative;
+}
+
 function getDocumentMetadata(url, window) {
   const doc = window.document;
   const metadata = getMetadata(doc);
 
-  metadata.url = url;
-  metadata.original_url = url;
-  metadata.provider_url = url;
-
-  metadata.favicon_url = metadata.icon_url;
-  if (!metadata.favicon_url) {
-    const parsedUrl = urlparse.parse(url);
-    metadata.favicon_url = `${parsedUrl.protocol}//${parsedUrl.host}/favicon.ico`;
-  }
-
-  metadata.image_url = metadata.image_url && metadata.image_url.replace(/^\/\//, 'https://');
-  metadata.images = [{
-    entropy: 1.0,
-    height: 500,
-    url: metadata.image_url,
-    width: 500,
-  }];
+  const responseData = {
+    url: url,
+    provider_url: url,
+    original_url: url,
+    title: metadata.title,
+    description: metadata.description,
+    favicon_url: metadata.icon_url ? makeUrlAbsolute(url, metadata.icon_url) : makeUrlAbsolute(url, '/favicon.ico'),
+    images: [{
+      url: makeUrlAbsolute(url, metadata.image_url),
+      width: 500,
+      height: 500,
+      entropy: 1.0,
+    }]
+  };
 
-  console.log(`Generated Metadata for ${url}:\n${JSON.stringify(metadata)}`); // eslint-disable-line no-console
+  console.log(`Generated Metadata for ${url}:\n${JSON.stringify(responseData)}`); // eslint-disable-line no-console
```

Comment:
Not sure if we want to only `console.log()` conditionally based on environment (see https://github.com/mozilla/page-metadata-service/pull/23).

Adding logging makes the Circle-CI logs a bit harder to read since you get metadata thrown in with the mocha output.